sparkmagic/sparkmagic/handlers.py

```diff
@@ -0,0 +1,95 @@
+import json
+from notebook.utils import url_path_join
+from notebook.base.handlers import IPythonHandler
+
+from sparkmagic.kernels.kernelmagics import KernelMagics
+from sparkmagic.utils.sparkevents import SparkEvents
+
+
+class ReconnectHandler(IPythonHandler):
+    def post(self):
+        path, username, password, endpoint = self.get_arguments()
+        spark_events = self.get_spark_events()
+        
+        # Get kernel manager
+        kernel_manager = self.get_kernel_manager(path)
+        if kernel_manager is None:
+            status_code = 404
+            self.set_status(status_code)
+            error = "No kernel for given path"
+            self.finish(json.dumps(dict(success=False, error=error)))
+            spark_events.emit_cluster_change_event(endpoint, status_code, False, error)
+            return
+
+        # Restart
+        kernel_manager.restart_kernel()
+
+        # Execute code
+        client = kernel_manager.client()
+        code = '%{} -s {} -u {} -p {}'.format(KernelMagics._do_not_call_change_endpoint.__name__, endpoint, username, password)    
+        response_id = client.execute(code, silent=False, store_history=False)
+        msg = client.get_shell_msg(response_id)
+
+        # Get execution info
+        successful_message = self.msg_successful(msg)
```

Comment:
That doesn't always return the same fields for the response. Why do you want this change?